### PR TITLE
#1638 - Move `archive application` from api to schedulers

### PR DESCRIPTION
**Pull request by andrepestana-aot**
- Created Process Archive Application Scheduler;
- Remove ApplicationSystem from API;

image

### CODE REVIEW

File: configs/env-example [hunk 1]
```diff
@@ -9,7 +9,7 @@ BUILD_ID=0
 API_PORT=3000
 APP_PORT=8080
 SWAGGER_ENABLED=true
-APPLICATION_ARCHIVE_DAY=43
+APPLICATION_ARCHIVE_DAYS=43
```
Comment (andrewsignori-aot) [inline configs/env-example:12]: Adding the comment here for the lack of a better place.
The env variable `APPLICATION_ARCHIVE_DAYS` must be moved from the `api-deploy.yml` to the `queue-consumers-deploy.yml`.

File: sources/packages/backend/apps/api/src/app.module.ts [hunk 2]
```diff
@@ -55,10 +52,6 @@ import { QueueModule } from "@sims/services/queue";
         path: ClientTypeBaseRoute.SupportingUser,
         module: AppSupportingUsersModule,
       },
-      {
-        path: ClientTypeBaseRoute.SystemAccess,
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/app.module.ts:59]: `SystemAccess` can also be removed from the enum `ClientTypeBaseRoute`.

File: sources/packages/backend/apps/api/src/app.system-access.module.ts [hunk 3]
```diff
@@ -1,70 +0,0 @@
-// Todo: remove this module when archive application `/route-controllers/application/application.system-access.controller.ts` is moved to integartion.
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/app.system-access.module.ts:1]: This is the end of an era 😄

File: sources/packages/backend/apps/db-migrations/src/sql/Queue/Add-archive-applications-queue.sql [hunk 4]
```diff
@@ -0,0 +1,13 @@
+INSERT INTO
+  queue_configurations(queue_name, queue_configuration)
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/db-migrations/src/sql/Queue/Add-archive-applications-queue.sql:2]: Please use the schema also `sims.queue_configurations`.

File: sources/packages/backend/apps/queue-consumers/src/queue-consumers.module.ts [hunk 5]
```diff
@@ -23,18 +23,23 @@ import {
...
   ATBCResponseIntegrationScheduler,
+  ProcessArchiveApplicationsScheduler,
 } from "./processors";
 import {
   DisbursementScheduleService,
   SequenceControlService,
   StudentRestrictionSharedService,
   WorkflowClientService,
   ZeebeModule,
+  ApplicationService,
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/queue-consumers/src/queue-consumers.module.ts:34]: Can this service be at `queue-consumers` scope?
We have been adding to `@sims/services` services that potentially could be reused by multiple apps. This one seems to be a `queue-consumers` task for now.

**Review by andrewsignori-aot — COMMENTED** [review]
Nice work, please take a look at the comments.

File: sources/packages/backend/apps/db-migrations/src/sql/Queue/Rollback-add-archive-applications-queue.sql [hunk 6]
```diff
@@ -0,0 +1,4 @@
+DELETE FROM
+  sims.queue_configurations
+WHERE
+  queue_name = 'process-archive-applications';
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/db-migrations/src/sql/Queue/Rollback-add-archive-applications-queue.sql:4]: I would recommend the queue name to be just `archive-applications`.

File: sources/docker-compose.yml [hunk 7]
```diff
@@ -149,6 +149,7 @@ services:
       - ESDC_REQUEST_FOLDER=${ESDC_REQUEST_FOLDER}
       - ESDC_RESPONSE_FOLDER=${ESDC_RESPONSE_FOLDER}
       - ESDC_ENVIRONMENT_CODE=${ESDC_ENVIRONMENT_CODE}
+      - APPLICATION_ARCHIVE_DAYS=${APPLICATION_ARCHIVE_DAYS}
```
Comment (guru-aot) [inline sources/docker-compose.yml:152]: can we remove APPLICATION_ARCHIVE_DAYS it from api environments? and also remove from api-deploy.yml?

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/application/process-archive-application.scheduler.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  @Process()
+  async processArchiveApplications(
+    job: Job<void>,
+  ): Promise<QueueProcessSummaryResult> {
+    const summary = new QueueProcessSummary({
+      appLogger: this.logger,
+      jobLogger: job,
+    });
+    await summary.info("Processing pending applications to be archived.");
+
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/application/process-archive-application.scheduler.ts:42]: Please remove the blank line here.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/application/process-archive-application.scheduler.ts [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    const auditUser = await this.systemUsersService.systemUser();
+    const archivedApplicationsCount =
+      await this.applicationService.archiveApplications(auditUser.id);
+    await summary.info(
+      `Total of applications archived: ${archivedApplicationsCount}`,
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/queue-consumers/src/processors/schedulers/application/process-archive-application.scheduler.ts:47]: Total count | Total number

**Review by dheepak-aot — APPROVED** [review]
Thanks for getting rid of System access components. LGTM 👍
Added 2 comments which are very minor and optional.

File: sources/packages/backend/apps/queue-consumers/src/processors/schedulers/application/process-archive-application.scheduler.ts (hunk 9 above)
Comment (andrepestana-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/application/process-archive-application.scheduler.ts:47]: I didn't get what you mean? Is this about the text or variable name?

Comment (dheepak-aot) [reply sources/packages/backend/apps/queue-consumers/src/processors/schedulers/application/process-archive-application.scheduler.ts:46]: The text.

**Review by guru-aot — APPROVED** [review]
LGTM, thanks for doing the changes. Nice work @andrepestana-aot

**Pull request merged**